cli/listener/src/main.rs

```diff
@@ -394,7 +402,169 @@ impl Metrics {
+        error: "Missing key for second parameter, should be \"from\" or \"to\"".to_owned(),
+    })?;
+    let date_second_value = param_format.next().ok_or(ResponseError {
+        error: "Missing date value for second parameter".to_owned(),
+    })?;
+    match (first_key, second_key) {
+        ("from", "to") => Ok(DateFromTo{
+          from: parse_utc_iso8601(date_first_value),
+          to:   parse_utc_iso8601(date_second_value)
+        }
+        ),
+        ("to", "from") => Ok(DateFromTo{
+          from: parse_utc_iso8601(date_second_value),
+          to:   parse_utc_iso8601(date_first_value),
+        }
+        ),
+        _ => Err(ResponseError {
+            error: format!("Invalid parameters, should be \"to\" or \"from\" in any order, is {} and {} instead", first_key, second_key)
+        }),
+    }
+}
+
+fn get_error_response(err_res: ResponseError) -> ResponseBox {
+    let content_type = Header::from_bytes(&b"Content-Type"[..], &b"application/json"[..])
+        .expect("Static header value, does not fail at runtime.");
+    Response::from_data(serde_json::to_vec(&err_res).expect("Serialization shouldn't fail"))
+        .with_status_code(500)
+        .with_header(content_type)
+        .boxed()
+}
+
+/// Get an interval price, consume it and returns a `ResponseBox` with the
+/// provided interval price and computed annual percentage rate.
+fn get_success_response(interval_prices: IntervalPrices) -> ResponseBox {
+    let response_interval = ResponseInterval {
+        annual_percentage_rate: interval_prices.annual_percentage_rate(),
+        interval_prices,
+    };
+    let content_type = Header::from_bytes(&b"Content-Type"[..], &b"application/json"[..])
+        .expect("Static header value, does not fail at runtime.");
+    Response::from_data(
+        serde_json::to_vec(&response_interval).expect("Serialization shouldn't fail"),
+    )
+    .with_header(content_type)
+    .boxed()
+}
+
+/// Gets a response that will be sent to a requesting client.
+/// Returns None if no method is passed.
+fn get_interval_price_request(
+    db_connection: &Connection,
+    request: &Request,
+) -> Option<ResponseBox> {
+    let mut request_split = request.url().split('/');
+    // Ignore first string.
+    request_split.next()?;
+    let parameters_result = if let Some(method) = request_split.next() {
+        let vec_parameters = method.split('?').collect::<Vec<&str>>();
+        let method_name = *vec_parameters.get(0)?;
+        if method_name == "interval_price" {
+            match get_date_params(&vec_parameters[1..]) {
+                Ok(res) => Some(res),
+                Err(err) => return Some(get_error_response(err)),
+            }
+        } else if method_name.is_empty() {
+            None
+        } else {
+            return Some(get_error_response(ResponseError {
+                error: format!("Method not supported: {}, use \"interval_price?from_date=<from_date_iso8601>?to_date=<to_date_iso8601>\"", method_name)
```

Comment:
Query params tend to have `?` to separate the path part from the query params, and then `&` to separate different query params, not a second `?`, e.g.

```
/interval_price?from_date=X&to_date=Y
```